doc/algorithms.rst

```diff
@@ -218,6 +219,11 @@ By specifying a lambda expression that takes a :cpp:`phlex::handle<hits>` object
 
 The lambda expression *does* depend on framework interface; the :cpp:`make_tracks_debug` function, however, retains its framework independence.
 
+.. admonition:: Chris Green
+   :class: admonition-chg
+
+   If one passes a framework-dependent closure to an algorithm, is that algorithm still framework-independent?
```

Comment:
To pass a closure to an algorithm, that algorithm must be a template.  If the algorithm template can be placed in a header file with no framework dependencies, then the template is framework-independent.  The instantiation, however, occurs wherever the algorithm would be registered--and that would certainly be framework dependent.

That said, I don't expect most users to pass closures to algorithms.

Reply:
Understood, thank you. IIRC one can pass a closure as an `std::function`, which would allow a C++ algorithm using such to be not a template, no?

Reply:
>  IIRC one can pass a closure as an std::function, which would allow a C++ algorithm using such to be not a template, no?

Sure, but the dependency has to do with the interface. (Maybe we should make that explicit?)

An algorithm might look like:

```c++
auto make_tracks(std::function<Tracks(Hits const&)> track_maker, ...);
```

where `Hits` and `Tracks` know nothing of the framework.  But maybe it's then called like:

```c++
auto phlex_thingy = phlex::get_something(...);
make_tracks(
  [&phlex_thingy](Hits const& hs) { return phlex_thingy(hs); }
);
```

In this case, the *invocation* of `make_tracks` depends on the framework, but the algorithm implementation does not.

Reply:
I'm OK with that: it gives authors the ability to be independent of the framework, while still allowing certain information to be provided by the framework when appropriate, and by something else otherwise.